common/djangoapps/track/eventshim.py

```diff
@@ -0,0 +1,380 @@
+    """
+    Shim to maintain backward compatiblity with seq_goto events.
+    """
+
+    shim_name = u'edx.ui.lms.sequence.tab_selected'
+    is_legacy_event = True
+    legacy_event_type = u'seq_goto'
+
+    def process_legacy_fields(self):
+        self.event[u'old'] = self.event[u'current_tab']
+        self.event[u'new'] = self.event[u'target_tab']
+
+
+class _BaseLinearSequenceEventShim(EventShim):
+    """
+    Common functionality for shimming
+    `edx.ui.lms.sequence.{next,previous}_selected` events.
+    """
+
+    offset = None
+
+    @property
+    def is_legacy_event(self):
+        return not self.crosses_boundary()
+
+    def process_legacy_fields(self):
+        self.event[u'old'] = self.event[u'current_tab']
+        self.event[u'new'] = self.event[u'current_tab'] + self.offset
+
+    def crosses_boundary(self):
+        raise NotImplementedError
+
+
+@EventShimRegistry.register
+class NextSelectedEventShim(_BaseLinearSequenceEventShim):
+    """
+    Shim to maintain backward compatiblity with seq_next events.
+    """
+
+    shim_name = u'edx.ui.lms.sequence.next_selected'
+    offset = 1
+    legacy_event_type = u'seq_next'
+
+    def crosses_boundary(self):
+        return self.event[u'current_tab'] == self.event[u'tab_count']
+
+
+@EventShimRegistry.register
+class PreviousSelectedEventShim(_BaseLinearSequenceEventShim):
+    """
+    Shim to maintain backward compatiblity with seq_prev events.
+    """
+
+    shim_name = u'edx.ui.lms.sequence.previous_selected'
+    offset = -1
+    legacy_event_type = u'seq_prev'
+
+    def crosses_boundary(self):
+        return self.event[u'current_tab'] == 1
+
+
+@EventShimRegistry.register
+class VideoEventShim(EventShim):
+    """
+    Converts new format video events into the legacy video event format.
+
+    Mobile devices cannot actually emit events that exactly match their
+    counterparts emitted by the LMS javascript video player. Instead of
+    attempting to get them to do that, we instead insert a shim here that
+    converts the events they *can* easily emit and converts them into the
+    legacy format.
+
+    TODO: Remove this shim and perform the conversion as part of some batch
+    canonicalization process.
+    """
+    shim_name = u'edx.video.'
+
+    name_to_event_type_map = {
+        u'edx.video.played': u'play_video',
+        u'edx.video.paused': u'pause_video',
+        u'edx.video.stopped': u'stop_video',
+        u'edx.video.loaded': u'load_video',
+        u'edx.video.position.changed': u'seek_video',
+        u'edx.video.seeked': u'seek_video',
+        u'edx.video.transcript.shown': u'show_transcript',
+        u'edx.video.transcript.hidden': u'hide_transcript',
+    }
+
+    is_legacy_event = True
+
+    @property
+    def legacy_event_type(self):
+        return self.name_to_event_type_map[self.name]
+
+    def process_event(self):
+        if self.name not in self.name_to_event_type_map:
+            return
+        # import pdb; pdb.set_trace()
```

Comment:
I think the commented-out code here should be deleted or clarified if it is informative.